ci/all_tests.sh

```diff
@@ -88,7 +88,7 @@ for roc_file in $EXAMPLES_DIR*.roc; do
         $absolute_roc dev $base_file $ROC_BUILD_FLAGS
         cd ..
     else
-        $ROC dev $roc_file $ROC_BUILD_FLAGS
+        $ROC dev --linker=legacy --prebuilt-platform $roc_file $ROC_BUILD_FLAGS
```

Comment:
Is there a tracking issue for being able to remove this `--linker=legacy` option (and the one above on line 38)? If so, can we put a comment with a TODO linking to said ticket?

Reply:
This is a good point. I think with this change, now that 6806 has landed, we could revisit this and test also the surgical prebuilt binaries. 

Also while we are at it, we could run the LLVM and DEV backends. 

Something to discuss further... maybe that would just be overkill.

Reply:
I suspect we would want to just test using the surgical binaries for the supported arch's, that will be orders of magnitude faster to run in CI. The purpose of these tests is to test basic-cli, and not roc so I don't see why we need to test LLVM here.